plugins/aws/aws_vault_importer.go

```diff
@@ -0,0 +1,87 @@
+package aws
+
+import (
+	"context"
+	"os"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+	"github.com/99designs/aws-vault/v7/cli"
+	"github.com/99designs/aws-vault/v7/vault"
+)
+
+// TryAwsVaultCredentials looks for the access key in the user's vaulting backend through AWS Vault.
+func TryAwsVaultCredentials() sdk.Importer {
+	return TryAWSVault(func(ctx context.Context, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		// Determine the vaulting backend through AWS_VAULT_BACKEND or based on the OS
+		awsVault := &cli.AwsVault{}
+		if awsVaultBackend := os.Getenv("AWS_VAULT_BACKEND"); awsVaultBackend != "" {
+			awsVault.KeyringBackend = awsVaultBackend
+		}
+		keyring, err := awsVault.Keyring()
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+
+		// Use the CredentialKeyring struct from aws-vault to retrieve vaulting backend credentials
+		credentialKeyring := &vault.CredentialKeyring{Keyring: keyring}
+
+		// Load the AWS config file (default location at ~/.aws/config)
+		awsConfigFile, err := awsVault.AwsConfigFile()
+		if err != nil {
+			out.AddError(err)
+			return
+		}
+		configLoader := &vault.ConfigLoader{File: awsConfigFile}
+
+		// Get the region specified for the "default" profile
+		var defaultRegion string
+		if defaultSection, ok := awsConfigFile.ProfileSection("default"); ok {
+			defaultRegion = defaultSection.Region
+		}
+
+		// Iterate through the profiles in the AWS config file and
+		// import any matching credentials stored in the vaulting backend
+		for _, profileName := range awsConfigFile.ProfileNames() {
+			if profileName != "default" {
+				creds, err := credentialKeyring.Get(profileName)
+				if err == nil {
```

Comment:
Let's make sure we are adding this error to the output's diagnostics if it is not nil. If a user specified a profile in their config file, they would expect to see an error if that profile's credentials are not to be found in the keyring.

Reply:
Thanks for catching that, done! FYI I made it so that the code doesn't `return` if fetching credentials for one profile fails, since it's possible that fetching credentials for other profiles that come later in the config file could still succeed.

By the way, currently, errors during importing/provisioning display like this:

![image](https://user-images.githubusercontent.com/53918631/229221778-78f85286-a87d-4cab-9224-0b4d82d2b134.png)

Should we make it so that errors display in a more user-friendly way? This would be a change in the 1Password CLI code, from my understanding.

Reply:
On a second consideration, I realised that we should rather treat a not found secret as we currently do for other credentials we are searching for in the import step: if we don't find them, just don't include them in the candidate list. What this means in our case is to check in this block whether or not the error that we get is the `credentials not found` one and in that case do not show an error, but rather simply skip these creds:

```
creds, err := credentialKeyring.Get(profileName)
 			if err != nil && strings.Contains(err.Error(), "not found") {
 				out.AddError(fmt.Errorf("could not retrieve credentials for %s from vaulting backend %s: %w", profileName, string(awsVaultBackend), err))
```